tests/components/onewire/test_config_flow.py

```diff
@@ -0,0 +1,263 @@
+"""Tests for 1-Wire config flow."""
+from pyownet import protocol
+
+from homeassistant.components.onewire.const import (
+    CONF_MOUNT_DIR,
+    CONF_TYPE_OWSERVER,
+    CONF_TYPE_SYSBUS,
+    DEFAULT_OWSERVER_PORT,
+    DEFAULT_SYSBUS_MOUNT_DIR,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_TYPE
+from homeassistant.data_entry_flow import (
+    RESULT_TYPE_ABORT,
+    RESULT_TYPE_CREATE_ENTRY,
+    RESULT_TYPE_FORM,
+)
+
+from . import setup_onewire_owserver_integration, setup_onewire_sysbus_integration
+
+from tests.async_mock import patch
+
+
+async def test_user_owserver(hass):
+    """Test OWServer user flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        user_input={CONF_TYPE: CONF_TYPE_OWSERVER},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "owserver"
+    assert result["errors"] == {}
+
+    # Invalid server
+    with patch(
+        "homeassistant.components.onewire.config_flow.protocol.proxy",
+        side_effect=protocol.ConnError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={CONF_HOST: "1.2.3.4", CONF_PORT: 1234},
+        )
+
+        assert result["type"] == RESULT_TYPE_FORM
+        assert result["step_id"] == "owserver"
+        assert result["errors"] == {"base": "cannot_connect"}
+
+    # Valid server
+    with patch(
+        "homeassistant.components.onewire.config_flow.protocol.proxy",
+    ):
```

Comment:
We should patch the integration setup functions, to avoid setting up the whole integration on create entry results. See our scaffold script template test code for example.

Reply:
I'm working on this one...

Reply:
New OneWireHub class created for this.
Is it ok to update __init__.py and sensor.py as a follow-up PR?

Reply:
I was only referring to this:
https://github.com/home-assistant/core/blob/93841e3e0cfe8971c322b6b30cb32b7cd4abd508/script/scaffold/templates/config_flow/tests/test_config_flow.py#L21-L25

Reply:
How to make things complicated! Should I keep the new OneWireHub (will be useful down the line with the binary-sensor and switch PRs) or remove it again to keep this PR smaller?

Reply:
Can we keep the OneWireHub class anyway or do you want me to remove it?
I plan to use it with followup PRs (binary-sensor / switch platforms)

Reply:
We can keep it.